### ISSUE TITLE
FIPS mode not working

### ISSUE THREAD

**Issue by ssk199441** [erlang/otp#6981]
**Describe the bug**
I have installed openssl 3.0.7 with FIPS mode enabled on alpine 3.17, i need erlang also to be FIPS compliance to run rabbitmq-server in FIPS mode. But i tried to enable fips mode in erlang it is not working 

**To Reproduce**
Steps to reproduce the behavior.
1. wget https://github.com/erlang/otp/releases/download/OTP-25.2.3/otp_src_25.2.3.tar.gz
2. tar -zxvf otp_src_25.2.3.tar.gz
3. ./otp_build configure --enable-fips
4. make and make install

**Expected behavior**

Trying to get output using 
erl -eval 'io:format("~s~n", [crypto:supports(fips)]).' -noshell 
It should return true/false but it throws error

**Error**
/otp_src_25.2.3 # erl -eval 'io:format("sn", [crypto:supports(fips)]).' -noshell
{"init terminating in do_boot",{function_clause,[{crypto,supports,[fips],[{file,"crypto.erl"},{line,570}]},{erl_eval,do_apply,7,[{file,"erl_eval.erl"},{line,744}]},{erl_eval,expr,6,[{file,"erl_eval.erl"},{line,282}]},{erl_eval,expr_list,7,[{file,"erl_eval.erl"},{line,961}]},{erl_eval,expr,6,[{file,"erl_eval.erl"},{line,454}]},{init,start_it,1,[]},{init,start_em,1,[]},{init,do_boot,3,[]}]}}
init terminating in do_boot ({function_clause,[{crypto,supports,[fips],[{_},{_}]},{erl_eval,do_apply,7,[{_},{_}]},{erl_eval,expr,6,[{_},{_}]},{erl_eval,expr_list,7,[{_},{_}]},{erl_eval,expr,6,[{_},{_}]},{init,start_it,1,[]},{init,start_em,1,[]},{init,do_boot,3,[]}]})

Crash dump is being written to: erl_crash.dump...done

**Comment by ssk199441** [comment]
Could you please send the steps for how you built in RHEL?

**Comment by nickva** [comment]
@ssk199441 the build is part of a set of CouchDB CI images, we build them for Debian, Ubuntu, and RHEL-like clones https://github.com/apache/couchdb-ci/commit/74e7543694f169672bf8fc00f326e76a5294b7d9

**Comment by KennethL** [comment]
FIPS mode is not yet supported in combination with OpenSSL 3.0

**Comment by nickva** [comment]
@yonashaile this is what I get on Rocky 8 (RHEL 8 clone):

```
Eshell V12.3.2.9  (abort with ^G)

> application:load({application, crypto, [{env, [{fips_mode, true}]}]}), application:start(crypto).
ok

> enabled = crypto:info_fips().
enabled

> crypto:info_lib().
[{<<"OpenSSL">>,269488319,
  <<"OpenSSL 1.1.1k  FIPS 25 Mar 2021">>}]
```

**Comment by ssk199441** [comment]
i have erlang 25.2.3 and openssl 3.0.7 installed on my alpine 3.17
when i try to run above commands  i am getting error
`
Eshell V13.1.5  (abort with ^G)
1> application:load({application, crypto, [{env, [{fips_mode, true}]}]}), application:start(crypto).
ok
2> enabled = crypto:info_fips().
=ERROR REPORT==== 7-Mar-2023::20:28:50.837493 ===
Unable to load crypto library. Failed with error:
"load, Library load-call unsuccessful (244)."

=WARNING REPORT==== 7-Mar-2023::20:28:50.839107 ===
The on_load function for module crypto returned:
{error,{load,"Library load-call unsuccessful (244)."}}

** exception error: undefined function crypto:info_fips/0
3> crypto:info_lib().
=ERROR REPORT==== 7-Mar-2023::20:29:22.150896 ===
Unable to load crypto library. Failed with error:
"load, Library load-call unsuccessful (244)."

** exception error: undefined function crypto:info_lib/0
4> =WARNING REPORT==== 7-Mar-2023::20:29:22.151577 ===
The on_load function for module crypto returned:
{error,{load,"Library load-call unsuccessful (244)."}}
 `

**Comment by sathesun** [comment]
When i can see the combination of openssl3.x and erlang works

**Comment by sverker** [comment]
Fixed by #7392.

**Comment by sverker** [comment]
The PR #7392 is based on OTP 26.0 and can be released in OTP-26.1 after the summer.
